giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/aspects/access/RestSecurityAspect.java

```diff
@@ -419,6 +489,42 @@ private void fillUser(User filled, User toBeFilled) {
         return new ResponseEntity<String>(sw.toString(), status);
     }
     
+    private String extractImagePath(ProceedingJoinPoint joinPoint) throws UnsupportedEncodingException {
+        Object[] args = joinPoint.getArgs();
+        MethodSignature sig = (MethodSignature) joinPoint.getSignature();
+        String[] argNames = sig.getParameterNames();
+        Class<?>[] argTypes = sig.getParameterTypes();
+        
+        String imagePath = null;
+        for (int i = 0; i < argNames.length; i++) {
+            if (HttpServletRequest.class.isAssignableFrom(argTypes[i])) {
+                Map<String, String[]> parameters = ((HttpServletRequest)args[i]).getParameterMap();
+                
+                StringBuffer parameterBuffer = new StringBuffer();
+                for (String key : parameters.keySet()) {
+                    for (String value : parameters.get(key)) {
+                        parameterBuffer.append(key);
+                        parameterBuffer.append("=");
+
+                        parameterBuffer.append(URLEncoder.encode(value, "UTF-8"));
+                        parameterBuffer.append("&");
```

Comment:
but you're not doing anything with that buffer after you've appended all these things, do you?